### PR TITLE
Improve daily test

### PR DESCRIPTION
**Pull request by Zelldon**
Due to suggestions by @saig0 I improved a bit the daily test.
It will now immediately run the test and trigger it again after one day. You can also retrigger it with a message.

Happy to discuss this @pihme 

P.s.: Not really sure whether I like that we need to always update the workflow images :sweat_smile:  Would like to have an automated thing or directly use of the xml.

### CODE REVIEW

**Comment by pihme** [comment]
Is this valid BPMN?
image

If so I would prefer it like this to avoid duplicating the input/output mapping

**Comment by pihme** [comment]
@Zelldon If we have a shell script that transforms the BPMN files to images, we probably can automated that part

**Comment by Zelldon** [comment]
> @Zelldon If we have a shell script that transforms the BPMN files to images, we probably can automated that part

Not sure how easy that is. Maybe we can embed somehow bpmn.io.

> If so I would prefer it like this to avoid duplicating the input/output mapping

Hm ok, but then probably non-interrupting.

**Comment by Zelldon** [comment]
@pihme maybe this helps https://github.com/bpmn-io/bpmn-to-image/

**Comment by Zelldon** [comment]
@saig0 any opinions on this?

I have also some other alternatives

Embedded subprocess to avoid duplicated input mappings with event sub processes

daily-test-protocol

OR Embedded Subprocess with event based gateway :D 

alternative

**Comment by saig0** [comment]
@Zelldon the process with the event-based gateway looks nice :laughing: 

I'm okay with the first version (main + 2 event subprocesses) or the one with the embedded subprocess. More than one start event for an event subprocess is not possible :sweat_smile: 

If you don't like the duplicated input mappings of the call activities then we could extract the call activity again into a new process 
(`run-daily-tests` vs `trigger-daily-tests`) :grin:

**Comment by Zelldon** [comment]
So why not going with the event-based gw?☺

**Comment by saig0** [comment]
:laughing:  The process with the event-based gateway looks a bit complicated. So, the intention of the process is not so clear (start every day or manually). But also the model has some semantical downsides:
* one test run blocks the next run, especially if the test failed (but I want that the tests run every day even if the analysis of the previous test run is not done)
* the process can be stopped only if no test run is active

**Comment by Zelldon** [comment]
@saig0 @pihme I updated the process to:

daily-test-protocol

If this is fine I will rebase and merge.

**Review by saig0 — APPROVED** [review]

File: workflows/daily-test-protocol.bpmn [hunk 1]
```diff
@@ -4,46 +4,98 @@
...
-      <bpmn:callActivity id="Activity_0ga0m8g" name="Run All Tests in Camunda Cloud per Cluster Plan">
+      <bpmn:outgoing>Flow_08as7sy</bpmn:outgoing>
+      <bpmn:subProcess id="Activity_01rcw4r" triggeredByEvent="true">
+        <bpmn:callActivity id="Activity_0ga0m8g" name="Run All Tests in Camunda Cloud per Cluster Plan">
+          <bpmn:extensionElements>
+            <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
+            <zeebe:ioMapping>
+              <zeebe:output source="=&#34;block variable propagation&#34;" target="dummy" />
+            </zeebe:ioMapping>
+            <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:18]:
```suggestion
            <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

File: workflows/daily-test-protocol.bpmn [hunk 2]
```diff
@@ -4,46 +4,98 @@
...
-      <bpmn:incoming>Flow_0jt1wdj</bpmn:incoming>
-    </bpmn:endEvent>
-    <bpmn:sequenceFlow id="Flow_0jt1wdj" sourceRef="Event_10m6wb5" targetRef="Event_16gkbgr" />
-    <bpmn:subProcess id="Activity_01rcw4r" triggeredByEvent="true">
-      <bpmn:callActivity id="Activity_0ga0m8g" name="Run All Tests in Camunda Cloud per Cluster Plan">
+      <bpmn:outgoing>Flow_08as7sy</bpmn:outgoing>
+      <bpmn:subProcess id="Activity_01rcw4r" triggeredByEvent="true">
+        <bpmn:callActivity id="Activity_0ga0m8g" name="Run All Tests in Camunda Cloud per Cluster Plan">
+          <bpmn:extensionElements>
+            <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:14]:
```suggestion
            <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

File: workflows/daily-test-protocol.bpmn [hunk 3]
```diff
@@ -4,46 +4,98 @@
...
+        <bpmn:sequenceFlow id="Flow_1d7hw5p" sourceRef="event-daily-trigger" targetRef="Activity_0ga0m8g" />
+        <bpmn:endEvent id="Event_0arjkp8">
+          <bpmn:incoming>Flow_0br4aao</bpmn:incoming>
+        </bpmn:endEvent>
+        <bpmn:sequenceFlow id="Flow_0br4aao" sourceRef="Activity_0ga0m8g" targetRef="Event_0arjkp8" />
+      </bpmn:subProcess>
+      <bpmn:subProcess id="Activity_0301x0i" triggeredByEvent="true">
+        <bpmn:callActivity id="Activity_03q5y87" name="Run All Tests in Camunda Cloud per Cluster Plan">
+          <bpmn:extensionElements>
+            <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:38]:
```suggestion
            <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

File: workflows/daily-test-protocol.bpmn [hunk 4]
```diff
@@ -4,46 +4,98 @@
...
+            <zeebe:ioMapping>
+              <zeebe:output source="=&#34;block variable propagation&#34;" target="dummy" />
+            </zeebe:ioMapping>
+            <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:42]:
```suggestion
            <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

File: workflows/daily-test-protocol.bpmn [hunk 5]
```diff
@@ -4,46 +4,98 @@
...
+        <bpmn:sequenceFlow id="Flow_1xbbcbp" sourceRef="Event_18keuqa" targetRef="Activity_03q5y87" />
+        <bpmn:startEvent id="Event_18keuqa" name="Run test" isInterrupting="false">
+          <bpmn:outgoing>Flow_1xbbcbp</bpmn:outgoing>
+          <bpmn:messageEventDefinition id="MessageEventDefinition_0pm1vdl" messageRef="Message_1w96d5v" />
+        </bpmn:startEvent>
+        <bpmn:association id="Association_1reid6x" sourceRef="Event_18keuqa" targetRef="TextAnnotation_0o41fut" />
+      </bpmn:subProcess>
+      <bpmn:callActivity id="Activity_102ufle" name="Run All Tests in Camunda Cloud per Cluster Plan">
         <bpmn:extensionElements>
           <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:60]:
```suggestion
          <zeebe:taskDefinition type="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

File: workflows/daily-test-protocol.bpmn [hunk 6]
```diff
@@ -4,46 +4,98 @@
...
           <zeebe:ioMapping>
-            <zeebe:input source="= &#34;new chaos&#34;" target="region" />
-            <zeebe:input source="=&#34;Internal Dev&#34;" target="channel" />
-            <zeebe:input source="=[&#34;Development&#34;, &#34;Production - S&#34;, &#34;Production - M&#34;, &#34;Production - L&#34;]" target="clusterPlans" />
-            <zeebe:input source="={&#34;steps&#34;:3,&#34;iterations&#34;:10,&#34;maxTimeForIteration&#34;:&#34;PT10S&#34;,&#34;maxTimeForCompleteTest&#34;:&#34;PT2M&#34;}" target="sequentialTestParams" />
             <zeebe:output source="=&#34;block variable propagation&#34;" target="dummy" />
           </zeebe:ioMapping>
           <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan-process" />
```
Comment (pihme) [inline workflows/daily-test-protocol.bpmn:64]:
```suggestion
          <zeebe:calledElement processId="run-all-tests-in-camunda-cloud-per-cluster-plan" />
```

**Review by pihme — APPROVED** [review]
Process names have changed in the meantime.

**Pull request merged**